package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/lua
+local site = require 'gluon.site_config'
+local uci = require('luci.model.uci').cursor()
+
+dnsmasq=uci:get_first("dhcp", "dnsmasq")
+
+uci:set('dhcp', dnsmasq, 'localise_queries', '1')
+uci:set('dhcp', dnsmasq, 'localservice', '0')
+
+if site.dns.servers then
+	uci:set('dhcp', dnsmasq, 'server', site.dns.servers)
+else
+	uci:delete('dhcp', dnsmasq, 'server')
+end
+
+if site.dns.cacheentries then
+	uci:set('dhcp', dnsmasq, 'cachesize', site.dns.cacheentries)
+else
+	uci:delete('dhcp', dnsmasq, 'cachesize')
+end
+
+if site.next_node.name then
+	if site.next_node.ip4 then
+		uci:section('dhcp','domain','nextnode4',{
+			name=site.next_node.name,
+			ip=site.next_node.ip4,
+		}
+		)
+	else
+		uci:delete('dhcp', 'domain', 'nextnode4')
+	end
+	if site.next_node.ip6 then
+		uci:section('dhcp','domain','nextnode6',{
+			name=site.next_node.name,
+			ip=site.next_node.ip6,
+		}
+		)
+	else
+		uci:delete('dhcp', 'domain', 'nextnode6')
+	end
+else
+	uci:delete('dhcp', 'domain', 'nextnode4')
+	uci:delete('dhcp', 'domain', 'nextnode6')
```

Comment:
Maybe get rid of the duplicate delete calls by changing the conditions to `site.next_node.name and site.next_node.ip4`, getting rid of the outer level of ifs.